pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -465,6 +465,12 @@ func (r *reconciler) currentLoadBalancerService(ci *operatorv1.IngressController
 		}
 		return false, nil, err
 	}
+
+	// BZ2054200: Don't delete services that are not directly owned by this controller
+	if service.Labels[manifests.OwningIngressControllerLabel] != ci.Name {
+		return false, nil, nil
```

Comment:
We should do something similar in `currentNodePortService` in `nodeport_service.go`.  

What happens if (1) the ingresscontroller has `spec.endpointPublishingStrategy.type: LoadBalancerService`, (b) a service with the desired name exists, and (c) the service isn't owned by the operator?

Reply:
Okay I'll add it in nodeport_service.go, good point. I was able to create another edge case (maybe this is what you were saying?) in which I create a ingress controller called "test" with `spec.endpointPublishingStrategy.type: LoadBalancerService` then created a service called `router-nodeport-test`. When the IngressController reconciles, it deletes `router-nodeport-test`, despite being a `LoadBalancerService` type.

This is because the nodeport_service.go reconciles regardless of `EndpointPublishingStrategy.type` and attempts to clean up unwanted services. Good catch.

Reply:
Added. Resolving this.